crates/prune/src/pruner.rs

```diff
@@ -375,48 +507,47 @@ impl<DB: Database> Pruner<DB> {
         to_block: BlockNumber,
         prune_mode: PruneMode,
     ) -> PrunerResult {
-        let from_block = provider
-            .get_prune_checkpoint(PrunePart::AccountHistory)?
-            .map(|checkpoint| checkpoint.block_number + 1)
-            .unwrap_or_default();
-        let range = from_block..=to_block;
-        let total = range.clone().count();
+        let range = match self.get_next_block_range_from_checkpoint(
+            provider,
+            PrunePart::AccountHistory,
+            to_block,
+        )? {
+            Some(range) => range,
+            None => {
+                trace!(target: "pruner", "No account history to prune");
+                return Ok(true)
+            }
+        };
+        let range_end = *range.end();
 
-        provider.prune_table_with_range_in_batches::<tables::AccountChangeSet>(
+        let mut last_pruned_block_number = None;
+        let (rows, done) = provider.prune_table_with_range::<tables::AccountChangeSet>(
             range,
             self.batch_sizes.account_history,
-            |keys, rows| {
-                trace!(
-                    target: "pruner",
-                    %keys,
-                    %rows,
-                    progress = format!("{:.1}%", 100.0 * keys as f64 / total as f64),
-                    "Pruned account history (changesets)"
-                );
-            },
+            |row| last_pruned_block_number = Some(row.0),
         )?;
+        trace!(target: "pruner", %rows, %done, "Pruned account history (changesets)");
 
-        self.prune_history_indices::<tables::AccountHistory, _>(
+        let last_pruned_block = last_pruned_block_number
```

Comment:
super nit: name is a bit confusing, maybe something like `let last_changeset_pruned_block` ? feel free to ignore lamo